driver/src/contracts/stablex_contract.rs

```diff
@@ -90,6 +106,34 @@ impl StableXContract for StableXContractImpl {
     }
 }
 
+fn parse_auction_data(packed_auction_bytes: Vec<u8>, index: U256) -> (AccountState, Vec<Order>) {
+    // extract packed auction info
+    assert_eq!(
+        packed_auction_bytes.len() % AUCTION_ELEMENT_WIDTH,
+        0,
+        "Each auction should be packed in 113 bytes"
+    );
+
+    let mut account_state = AccountState::default();
+    let mut order_count = HashMap::new();
+    let relevant_orders = packed_auction_bytes
```

Comment:
What a rust mastery!

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia0.giphy.com%2Fmedia%2FaLTFHi2aowjJe%2Fgiphy.gif&token=master" data-gitmeme-token="master"><img src="https://media0.giphy.com/media/aLTFHi2aowjJe/giphy.gif" title="Created by gitme.me with /master"/></a>

Reply:
Ben and I did this together, paircoding 👍 ! 

That is what happens after reviewing so many PRs from you 💃